From 53b501240e1f242a09fffa056e610441a01db56d Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 30 May 2019 00:32:59 +0200
Subject: [PATCH 266/386] telephony: mvebu_phone: document internal loopback
 validation tool

This patch extends the telephony documentation with
a description of newly added version of the mv_voice_tool,
capable of validating the TDM interface without external
connectivity.

Change-Id: I52a9cb850adceb605d374a428138ad740b47703e
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/10207
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../mvebu_phone/telephony_user_guide.txt      | 52 +++++++++++++++++++
 1 file changed, 52 insertions(+)

diff --git a/drivers/telephony/mvebu_phone/telephony_user_guide.txt b/drivers/telephony/mvebu_phone/telephony_user_guide.txt
index 8467019f566e..1ba7135491ee 100644
--- a/drivers/telephony/mvebu_phone/telephony_user_guide.txt
+++ b/drivers/telephony/mvebu_phone/telephony_user_guide.txt
@@ -181,3 +181,55 @@ Example for connecting 2 phones using mv-voice-tool::
 	>>  Please select option: 7
 	>>  Enter line #0: 1
 	>>  Enter line #1: 0
+
+generic mv-voice-tool
+---------------------
+User-space application which communicates via ioctl with the telephony driver.
+The application allows to configure the TDM interface TX/RX lines in
+internal loopback and check data integrity with sending/receiving incremental
+pattern. It doesn't require any external SLIC device and board modifications.
+
+In order to run mv-voice tool, run the following command::
+
+	# ./mv_voice_tool
+
+Example for connecting test 32 lines with 2-byte PCM sample size::
+
+	# ./mv_voice_tool
+
+	>>  Please enter total lines number: 32
+	>>  Please enter PCM sample size(1/2/4): 2
+
+	 !!! Remember to initialize TDM before running the test !!!
+
+	  Marvell Voice Tool:
+	  i. Initialize TDM device
+	  l. Internal loopback test
+	  d. Disable TDM device
+	  q. Quit
+
+	>>  Please select option: i
+	[  139.356516] mvebu_phone f47a0000.tdm: Loading Marvell Telephony Driver
+	[  139.363682] mvebu_phone f47a0000.tdm: enable internal loopback on TX/RX lines
+
+	  Marvell Voice Tool:
+	  i. Initialize TDM device
+	  l. Internal loopback test
+	  d. Disable TDM device
+	  q. Quit
+
+	>>  Please select option: l
+	>>  Enter number of iterations (minimum 4): 1000
+
+	Digital loopback test(32 lines) - PASS !!!
+
+	  Marvell Voice Tool:
+	  i. Initialize TDM device
+	  l. Internal loopback test
+	  d. Disable TDM device
+	  q. Quit
+
+	>>  Please select option: q
+
+	>>  TDM stopped
+	#
-- 
2.17.1

